### ISSUE TITLE
Refactor OneIT Intranet - Workflow Projects integration

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#5231]
As a maintainer of the Workflow API, we want to retire the endpoint /workflow/prod/api/developer/tools/forms/2183/packages because it performs terribly and will continue to get worse.

# Additional Information
This endpoint is used on https://git.uiowa.edu/its-web/data.its.uiowa.edu/-/blob/main/app/Http/Controllers/WorkflowController.php to create a feed of projects from Workflow that then are being consumed by the OneIT Intranet.

A few different options were provided and it was determined that keeping all items in the feed is preferred so the plan is to get access to the data warehouse and pull projects from there.

We have a service account that has been created to access this data. Credentials are in our password manager as well as the connection Url and Schema.

There is some additional information and example sql
https://iowa.sharepoint.com/sites/CSI-Drupal2/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FCSI%2DDrupal2%2FShared%20Documents%2FOneIT%20Intranet&viewid=bf54b9d9%2D0975%2D4fbc%2D9f1b%2D6377bdc2307c

**Comment by briand44** [comment]
@richardbporter you mentioned we are connecting to another database currently from data.its.uiowa.edu. It could be useful to point us at that as a reference?

**Comment by briand44** [comment]
>Once I’ve got things set up, I will send you a link to campus data where you can request access to our tables and optionally create a warehouse service account if you don’t already have one.

Marking this as blocked until ^^ is setup.

**Comment by richardbporter** [comment]
> it was determined that keeping all items in the feed is preferred

I assumed since the site is using Feeds, this was probably a requirement. You might be more familiar with that piece of things than me though.

Some DB docs exist at https://git.uiowa.edu/its-web/data.its.uiowa.edu#databases The credentials used for those DB connections should be stored in environment variables: https://git.uiowa.edu/its-web/data.its.uiowa.edu#environment-variables

Also, there is a data.its issue queue with a few items in it: https://git.uiowa.edu/its-web/data.its.uiowa.edu/-/issues This could be moved there and referenced in a card on the board, if desired.

**Comment by briand44** [comment]
>it was determined that keeping all items in the feed is preferred

>I assumed since the site is using Feeds, this was probably a requirement. You might be more familiar with that piece of things than me though.

Yeah, I agree I think that makes the most sense now as long as we don't have any performance issues getting all this data from the data warehouse. I think long term, possibly when the OneIt Intranet moves to v3, looking into the Workflow callbacks may make sense.

**Comment by joewhitsitt** [comment]
If we continue to use data.its, we might be blocked by this https://git.uiowa.edu/its-web/data.its.uiowa.edu/-/issues/21. Because of a recent Mac update, I am not sure our Vagrant/Virtualbox setup is working anymore which would make local development difficult.

**Comment by joewhitsitt** [comment]
>If we continue to use data.its...

Was authentication the original reason why data.its was used for this? Can using the Acquia secrets file allow us to re-build it within a module on SiteNow?

**Comment by briand44** [comment]
Closing as they are currently manually creating/updating these now and plan to move to SharePoint.
